File: Arrays & Hashing/1940. Longest Common Subsequence Between Sorted Arrays/1940.cpp
#pragma GCC optimze("O3","Oz","unroll-loops","inline-functions","fast-math","march=native","lto","")
class Solution {
public:
    vector<int> longestCommonSubsequence(vector<vector<int>>& arrays) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        int count[101]{};
        std::vector<int> res;
        
        for (int i = 0; i < arrays.size() - 1; i++) {
            for (auto& num : arrays[i]) {
                count[num]++;
            }
        }

        for (auto& num : arrays.back()) {
            if (count[num] == arrays.size() - 1) res.push_back(num);
        }

        return res;
    }
};

File: Arrays & Hashing/1940. Longest Common Subsequence Between Sorted Arrays/1940.py
class Solution:
    def longestCommonSubsequence(self, arrays: List[List[int]]) -> List[int]:
        res = arrays[0].copy()
        for i in range(1, len(arrays)):
            res = self.find_LCS(res, arrays[i])
        return res

        
    def find_LCS(self, arr1: List[int], arr2: List[int]) -> List[int]:
        arr1_counter = Counter(arr1)
        arr2_counter = Counter(arr2)

        ret = []

        for num, count in arr1_counter.items():
            if arr2_counter[num] == count:
                ret.append(num)
    
        return ret

        